lib/util.js

```diff
@@ -222,13 +223,21 @@ util = {
         return str;
     },
 
-    bufferOrArrayBufferToString: function (buffer) {
+    bufferOrArrayBufferToString: function (buffer, charset) {
         if (!buffer || _.isString(buffer)) {
             return buffer || '';
         }
-
-        var str = buffer.toString('utf8') || '';
-        (str === '[object ArrayBuffer]') && (str = util.arrayBufferToString(buffer));
+        var str = '';
+        try {
```

Comment:
flow incorrect.

this function provides same api for buffer and array buffer. if array buffer is detected first, we operate and bail out,